src/main/java/com/microsoft/jenkins/kubernetes/wrapper/V1ResourceManager.java

```diff
@@ -337,29 +420,89 @@ void notifyUpdate(V1Service original, V1Service current) {
         V1Job getCurrentResource() {
             V1Job job = null;
             try {
-                job = BATCH_V1_API_INSTANCE.readNamespacedJob(getName(), getNamespace(), getPretty(), true, true);
+                job = batchV1ApiInstance.readNamespacedJob(getName(), getNamespace(), getPretty(), true, true);
             } catch (ApiException e) {
-                handleApiException(e);
+                handleApiExceptionExceptNotFound(e);
             }
             return job;
         }
 
         @Override
         V1Job applyResource(V1Job original, V1Job current) {
             V1Job job = null;
+            V1Job putJob = getPutObject(original);
             try {
-                job = BATCH_V1_API_INSTANCE.replaceNamespacedJob(getName(), getNamespace(), current, getPretty(), null);
+                job = batchV1ApiInstance.replaceNamespacedJob(
+                        getName(), getNamespace(), putJob, getPretty(), null);
             } catch (ApiException e) {
                 handleApiException(e);
             }
             return job;
         }
 
+        /**
+         * Get PutObject with server-dryRun.
+         * Only some of the fields in the job resource can be modified.
+         * Directly putting the original object of the user will cause some immutable
+         * fields with defaultValue to be modified.
+         * This method is a workaround that populates the default value with the server dryRun implementation.
+         * https://kubernetes.io/blog/2019/01/14/apiserver-dry-run-and-kubectl-diff/
+         *
+         * @param original Current Object in Cluster
+         * @return Job With Default Value
+         */
+        V1Job getPutObject(V1Job original) {
+            // Clone Object to avoid modifications to the original object.
+            V1Job dryRunReq = KubernetesJsonUtils.getKubernetesJson().deserialize(
+                    KubernetesJsonUtils.getKubernetesJson().serialize(get()), V1Job.class);
```

Comment:
Change method to `getPutObject(V1Job original, V1Job current)` and replace `get()` with current passed by `applyResource`?